.github/workflows/create-release-pr.yaml

```diff
@@ -0,0 +1,357 @@
+            if [[ ! " ${failed_plugins[@]} " =~ " $plugin " ]]; then
+              failed_plugins+=("$plugin")
+            fi
+          done
+
+          for plugin in "${unit_test_plugins[@]}"; do
+            if [[ ! " ${failed_plugins[@]} " =~ " $plugin " ]]; then
+              failed_plugins+=("$plugin")
+            fi
+          done
+
+          for plugin in "${integration_test_plugins[@]}"; do
+            if [[ ! " ${failed_plugins[@]} " =~ " $plugin " ]]; then
+              failed_plugins+=("$plugin")
+            fi
+          done
+
+          echo "Failed plugins: ${failed_plugins[*]}"
+
+          # Get release for the branch name and docs
+
+          release_version="${{steps.current_available_version.outputs.current_available_version}}"
+          echo "Remote version = $release_version"
+
+          # Configure the git bot
+
+          git config --global user.name 'Release Bot'
+          git config --global user.email 'release-bot@users.noreply.github.com'
+
+          # Add all the changed files and then remove the failed plugins
+
+          git add .
+
+          for plugin in "${failed_plugins[@]}"; do
+            git restore --staged $plugin
+            git checkout -- $plugin
+          done
+
+          # Get the release notes and update the plugin decription with acapy version
+
+          body=$(python repo_manager.py 4)
+
+          # Update the RELEASES.md file with the release notes
+
+          echo "$body" | cat - RELEASES.md > temp && mv temp RELEASES.md
+
+          # Check if there are any upgrades
+          upgraded_plugins=$(python repo_manager.py 5)
+          has_upgrades=false
+
+          for plugin in ${potential_upgrades[@]}; do
+            for upgrade in ${upgraded_plugins[@]}; do
+              if [[ $plugin == *"$upgrade"* ]]; then
+                has_upgrades=true
+                break
+              fi
+            done
+          done
+
+          if [ "$has_upgrades" = true ]
+          then
+            echo "Upgrades detected. Committing the changes"
+          else
+            echo "No upgrades detected. Skipping commit and release"
+            exit 1
+          fi
+
+          # Update the release notes with the upgraded plugins
+
+          details="Plugins upgraded this release: "
+          for plugin in ${upgraded_plugins}; do
+            details="$details $plugin"
+          done
+
+          # Replace the first occurence of ' - ' with the details
+
+          sed -i "0,/\s\-\s/s// - ${details}/" RELEASES.md
+
+          # Add the updated pyproject.toml and RELEASES.md files
+          git add ./*pyproject.toml
+          git add ./RELEASES.md
+
+          git commit -s -m "Release v$release_version Upgrades"
+
+          # Prepare the PR body
+
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "pr_body<<$EOF" >> $GITHUB_OUTPUT
+          echo "$body $details" >> $GITHUB_OUTPUT
+          echo "$EOF" >> $GITHUB_OUTPUT
```

Comment:
I don't think the `git` operations are necessary here.  From what I understand that will all be handled by the `peter-evans/create-pull-request` in the next step.

Reply:
I think you may be right. I may be able to remove some of them. I was following this section https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#create-your-own-commits in the readme.

The thing was I am using git to remove the changes for the failed upgrades. I may be able to remove the add and commit statements at the end though.